scripts/db-heal-test.py

```diff
@@ -0,0 +1,238 @@
+
+    # Find working dir - Aida copies db-src
+    working_dir = max(Path(tmp_path).iterdir(), key=os.path.getmtime)
+    archive = working_dir / 'archive'
+    live = working_dir / 'live'
+
+    # Dumb carmen's logs into a file to avoid spamming
+    c = open(carmen_log_file, 'w')
+
+    # Restore Archive
+    result = subprocess.run(
+        ['go', 'run', './database/mpt/tool', 'reset', '--force-unlock', str(archive), str(latest_checkpoint)],
+        stdout=c,
+        stderr=c,
+        cwd=carmen_root)
+
+    check_program_failure(result.returncode, c)
+
+    # Export genesis to restore LiveDB
+    genesis = Path(tmp_path) / 'test_genesis.dat'
+
+    print(f"Restoration complete. Exporting LiveDB genesis block {latest_checkpoint}.")
+    result = subprocess.run(
+        ['go', 'run', './database/mpt/tool', 'export', '--block', str(latest_checkpoint), str(archive), str(genesis)],
+        stdout=c,
+        stderr=c,
+        cwd=carmen_root)
+    check_program_failure(result.returncode, c)
+
+    # Restore LiveDB
+    print("Export complete. Applying LiveDB genesis.")
+    try:
+        shutil.rmtree(live)
+        print("Live directory removed successfully.")
+    except FileNotFoundError:
+        print(f"Directory {live} does not exist.")
+        exit(1)
+    except PermissionError:
+        print(f"Permission denied to remove {live}.")
+        exit(1)
+    except Exception as e:
+        print(f"An error occurred: {e}")
+        exit(1)
+
+    print("Importing LiveDB genesis.")
+    result = subprocess.run(
+        ['go', 'run', './database/mpt/tool', 'import-live-db', str(genesis), str(live)],
+        stdout=c,
+        stderr=c,
+        cwd=carmen_root)
+    check_program_failure(result.returncode, c)
+
+    print(f"Iteration {i}/{number_of_iterations}")
+    # We restored to block X, although we need to start the app at +1 block because X is already done
+    first_block = latest_checkpoint + 1
+
+    print("Syncing restarted...")
+    command = [
+        './build/aida-vm-sdb', 'substate', '--validate',
+        '--db-tmp', tmp_path, '--carmen-schema', '5', '--db-impl', 'carmen',
+        '--aida-db', aida_db_path, '--no-heartbeat-logging', '--track-progress',
+        '--archive', '--archive-variant', 's5', '--archive-query-rate', '200',
+        '--carmen-checkpoint-interval', str(checkpoint_granularity), '--db-src',
+        str(working_dir), '--skip-priming', '--tracker-granularity',
+        str(checkpoint_granularity), str(first_block), str(last_block)
+    ]
+
+    os.chdir(aida_path)
+    with open(aida_log_file, 'w') as f:
+        process = subprocess.Popen(command, stdout=f, stderr=subprocess.STDOUT)
+    os.chdir(current_dir)
+
+    # Start monitoring the log file
+    latest_checkpoint = terminate_process_after(window, latest_checkpoint)
+
+    # Wait for the command to complete
+    process.wait()
+
+    if latest_checkpoint == -1:
+        os.chdir(current_dir)
+        sys.exit(1)
+
+    if last_working_dir:
```

Comment:
The reason why I delete it every iteration is because if the test fails, all of the data is kept there. This way only the latest db is prevailed

Reply:
but I guess it is because you terminate the loop (the whole script) by `sys.exit()`. If you `break` the loop, it would reach end of the script, where the temp directory can be deleted.

Reply:
I feet like there could be scenario where we would like to keep the broken db.